dataprofiler/profilers/profile_builder.py

```diff
@@ -1511,23 +1511,80 @@ def save(self, filepath=None):
+
+        :param data: Data to be profiled
+        :type data: Data class object
+        :param samples_per_update: Number of samples to use to generate profile
+        :type samples_per_update: int
+        :param min_true_samples: Min number of samples required for the profiler
+        :type min_true_samples: int
+        :param options: Options for the profiler.
+        :type options: ProfilerOptions Object
+        :param profiler_type: Type of Profiler ("structured"/"unstructured")
+        :type profiler_type: str
+        :return: BaseProfiler
+        """
+        # Attempt to construct according to user kwarg input
+        if profiler_type == "structured":
+            return StructuredProfiler(data, samples_per_update,
+                                      min_true_samples, options)
+        elif profiler_type == "unstructured":
+            return UnstructuredProfiler(data, samples_per_update,
+                                        min_true_samples, options)
+        elif profiler_type is not None:
+            raise ValueError("Must specify 'profiler_type' to be 'structured' "
+                             "or 'unstructured'.")
+
+        # If user doesn't specify, and data is Data object, use is_structured
+        if isinstance(data, data_readers.base_data.BaseData):
+            if data.is_structured:
+                return StructuredProfiler(data, samples_per_update,
+                                          min_true_samples, options)
+            else:
+                return UnstructuredProfiler(data, samples_per_update,
+                                            min_true_samples, options)
+
+        # If user doesn't specify, and data is Pandas object, need to infer
+        if isinstance(data, pd.DataFrame) or isinstance(data, pd.Series):
+            # If empty, default to structured
+            if not len(data):
+                warnings.warn("Empty data given to Profiler, will assume data "
+                              "is structured and create StructuredProfiler")
+                return StructuredProfiler(data, samples_per_update,
+                                          min_true_samples, options)
+
+            # If multi column df, data is structured
+            if isinstance(data, pd.DataFrame) and len(data.columns) > 1:
+                return StructuredProfiler(data, samples_per_update,
+                                          min_true_samples, options)
+
+            # If single column, look at string length and variance
+            data_ser = data if isinstance(data, pd.Series) else data[0]
+            if data_ser.dtype == "object":
+                sample = data_ser.sample(min(5, len(data_ser)))
+                sample_lens = sample.apply(len)
+                len_mean = sample_lens.mean()
+                len_range = sample_lens.max() - sample_lens.min()
+                # If strings are very long or varied, use unstructured
+                if len_mean > 25 or len_range > 25:
```

Comment:
Currently this is how I deem if a list of strings is "unstructured". I figured this catches long strings that probably couldn't be labelled all on their own or if the strings have varying lengths, since most structured labels are usually not too varied in length.

Reply:
Im still thinking about if this determination is appropriate, but in either case, I think the bar should be a little lower, like 15 characters or something

Reply:
What do you think the standard length for an address would be or full name? 
If it is small, I don't really see an issue as much for profiling it as structured. Thoughts?

Reply:
maybe just len_range. Not sure len_mean is needed.

Reply:
in general, if we have a single col with strings, it'd probably good to consider unstructured profile imo, unless users specify the type.

Reply:
@gme5078 @lettergram thoughts?

Reply:
I figured len_mean would cover the case where the range is small but all the strings are extremely long (like a bunch of same-length paragraphs or sentences)